### PR TITLE
[Filebeat] Fix reference leak in TCP and Unix socket inputs

### PR DESCRIPTION
**Pull request by andrewkroh**
## What does this PR do?

The tcp and unix input sources were leaking references causing a memory leak.
When an accepted connection ended inputsource/common.Closer was
supposed to delete the pointer that it held to the connection, but due to a code
error `delete` was being called on the wrong map.

## Why is it important?

This fixes a memory leak with the tcp and unix inputs.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by urso** [comment]
Good find. Please backport to 7.8 as well before FF.

**Comment by andrewkroh** [comment]
@urso Can you please take another look. My fix triggered race conditions so I took a different approach and dropped in a cancellable context in place of the common.Closer. https://github.com/elastic/beats/pull/19459/commits/d77ab50ca577c11a31d8a46e32bc3d7ab7c22e3e

File: filebeat/inputsource/common/listener.go [hunk 1]
```diff
@@ -110,13 +114,17 @@ func (l *Listener) run() {
 		}
 
 		handler := l.handlerFactory(*l.config)
-		closer := WithCloser(l.closer, func() { conn.Close() })
+		ctx, cancel := context.WithCancel(l.ctx)
+		go func() {
+			<-ctx.Done()
+			conn.Close()
+		}()
```
Comment (urso) [inline filebeat/inputsource/common/listener.go:121]: The handler go-routine is started always. Let's move cancellation setup local to the that go-routine. Consider to use `ctxtool.WithFunc`.

Comment (andrewkroh) [reply filebeat/inputsource/common/listener.go]: Nice, I updated it to use `ctxtool.WithFunc`.

**Comment by ph** [comment]
fyi @vjsamuel this might affect you?

**Comment by vjsamuel** [comment]
Thanks. I will need to redo our internal input during the next rebase. Moving to context is the right thing to do however.

**Pull request merged**